distributed/scheduler.py

```diff
@@ -5468,6 +5482,17 @@ async def add_client(self, comm: Comm, client: str, versions: dict) -> None:
         We listen to all future messages from this Comm.
         """
         parent: SchedulerState = cast(SchedulerState, self)
+
+        if parent._validate:
+            try:
+                stimulus_id = STIMULUS_ID.get()
+                raise AssertionError(
```

Comment:
This was an artefact of the previous PR based on `default_stimulus_id`. Previously, it was necessary to prevent  stimuli being set prior to `add_worker` and `add_client` rpc messages as these initiated handling loops like `handle_worker` for e.g. If these loops received a message `default_stimulus_id` would not override the existing stimuli. So this was a safety check.

Now that the handlers exlicitly *expect* stimuli to be provided (and can therefore override the existing stimulus), this is no longer a problem